### PR TITLE
Tweaks the powerfist

### PR DESCRIPTION
**Pull request by Unit2E**
## About The Pull Request

Changes the powerfist to not have a throw speed that normal walking people can match, rather actually flinging the person hit rather hard. Throw speed is based on the fist's setting. Also makes the powerfist leak its contents, based on the gas it contains and the settings.
(first, small, PR, be gentle)

## Why It's Good For The Game

Makes it so the puncher can't literally collide with people they punched, knocking themselves over, or other such shenanigans. Also makes the powerfist a lot more unique beyond just 'punches hard when you add gas', and clarifies some messages.

The launch speed might be a bit strong, and might warrant an up to the TC price. If I should do that, whack me with a stick about it.

## Changelog
:cl: Unit2E & JMoldy

balance: the powerfist now punches people away at high velocity depending on setting.
add: the powerfist now leaks the gas it uses onto the tile you're on.
tweak: Adds weak punch variations for empty and near-empty punching.

/:cl:

### CODE REVIEW

**Comment by Nickkoranda** [comment]
trips check em

**Comment by necromanceranne** [comment]
_SMASH_

### A NEW RECORD!!!

**Comment by nemvar** [comment]
So does that mean abusing that the thing only consumes moles now freezes you?

**Comment by Unit2E** [comment]
It does, depending on what you're wearing I suppose. If the content of the tank you're using is extremely cold, you'll leak extremely cold gas on the tile. Or if you're punching with hot plasma, you might start a plasma fire by punching someone.

**Comment by necromanceranne** [comment]
Punch so hard I cause a plasma fire? This is a buff.

**Comment by PKPenguin321** [comment]
This is pretty cool, we need more stuff that sends people flying

**Comment by Adanps** [comment]
Plasma Fist martial art can send people flying, but it has some... issues...

File: code/game/objects/items/powerfist.dm [hunk 1]
```diff
@@ -71,26 +70,42 @@
...
+    playsound(loc, 'sound/weapons/genhit2.ogg', 50, 1)
 
-	var/atom/throw_target = get_edge_target_turf(target, get_dir(src, get_step_away(target, src)))
+    var/atom/throw_target = get_edge_target_turf(target, get_dir(src, get_step_away(target, src)))
 
-	target.throw_at(throw_target, 5 * fisto_setting, 0.2)
+    target.throw_at(throw_target, 5 * fisto_setting, 0.5 + (fisto_setting / 2))
 
-	log_combat(user, target, "power fisted", src)
+    log_combat(user, target, "power fisted", src)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/powerfist.dm]: lewd

**Pull request merged**